### PR TITLE
chore(sequencer): add instrumentation

### LINKED ISSUE #1321: chore: add instrumentation on all async calls
Mentioned in the Protocol Weekly on 7/25, instrumentation should be added on all async calls.

Things to keep in mind (via Janis):

> 1. we need to remove instrumentation from all long-running tasks (in our case, essentially everything that is called T::run or T::run_until_stopped
> 2. we also need to remove all events that run under these (so if you have a loop { select!((....)} construct, then there cannot be events in these
>             a. The reason is that forever spans are completely useless on an observability platforms; in the best case we will see them at the end of a run, in the worst case we will lose them completely because the gRPC payload is too big (because the span contains too many events).
> 3. while we want to have #[instrument(skip_all, err)], we don't necessarily want the errors to be emitted under an ERROR target. ERROR is for service-stopping issues, while WARN is for issues that can be handled (for example, through a retry).
>             a. using this example, this would be #[instrument(err(level = Level::WARN)]

```
### Tasks
- [x] bridge-withdrawer
- [x] composer
- [x] conductor
- [ ] sequencer
- [x] sequencer-relayer
```

┆Issue Number: ENG-670

### PR DESCRIPTION
**Pull request by ethanoroshiba**
## Summary
Added instrumentation to most async functions which did not have it.

## Background
Adding instrumentation to all async calls will aid in tracing since spans will be emitted even if no events happen under them.

## Changes
- Added instrumentation to most async functions which did not have it.
- Added `err` argument to the `instrument` macro where applicable.
- Added fields to the `instrument` macro where applicable.

## Testing
Passing all tests.

## Changelogs
No updates required

## Related Issues
closes #1321

### CODE REVIEW

**Review by Fraser999 — CHANGES_REQUESTED** [review]
Blocking this for now (as much a reminder to myself as anything else).

Discussed offline with @ethanoroshiba that this will be revisited to ensure we're only using error-level instrumentation where it's warranted (i.e. where it indicates a serious problem or bug in the logic) and not for non-critical cases (e.g. handling a malformed client request).

File: crates/astria-sequencer/src/mempool/transactions_container.rs [hunk 1]
```diff
@@ -569,6 +573,7 @@ pub(super) trait TransactionsContainer<T: TransactionsForAccount> {
     /// Recosts transactions for an account.
     ///
     /// Logs an error if fails to recost a transaction.
+    #[instrument(skip_all, fields(address = %hex::encode(address)))]
```
Comment (Fraser999) [inline crates/astria-sequencer/src/mempool/transactions_container.rs:576]: I think the address should be base64-encoded.

File: crates/astria-sequencer/src/mempool/mod.rs [hunk 2]
```diff
@@ -482,7 +485,7 @@ impl Mempool {
     /// does not take into account gapped nonces in the parked queue. For example, if the
     /// pending queue for an account has nonces [0,1] and the parked queue has [3], [1] will be
     /// returned.
-    #[instrument(skip_all)]
+    #[instrument(skip_all, fields(address = %hex::encode(address)))]
```
Comment (Fraser999) [inline crates/astria-sequencer/src/mempool/mod.rs:488]: Address should be base64-encoded I think.

File: crates/astria-sequencer/src/sequencer.rs [hunk 3]
```diff
@@ -57,15 +58,19 @@ impl Sequencer {
             .try_exists()
             .context("failed checking for existence of db storage file")?
         {
-            info!(
-                path = %config.db_filepath.display(),
-                "opening storage db"
-            );
+            info_span!("Sequencer::run_until_stopped").in_scope(|| {
```
Comment (Fraser999) [inline crates/astria-sequencer/src/sequencer.rs:61]: We could just hold the span in a variable and re-use it throughout this method:
```rust
        let span = info_span!("Sequencer::run_until_stopped");
        let db_exists = config
            .db_filepath
            .try_exists()
            .wrap_err("failed checking for existence of db storage file")?;
        span.in_scope(|| {
            info!(path = %config.db_filepath.display(), "{} storage db", if db_exists {
                "opening"
            } else {
                "creating"
            });
        });
```

File: crates/astria-sequencer/src/sequencer.rs [hunk 4]
```diff
@@ -119,27 +124,29 @@ impl Sequencer {
             .wrap_err("failed to parse grpc_addr address")?;
         let grpc_server_handle = start_grpc_server(&storage, mempool, grpc_addr, shutdown_rx);
 
-        info!(config.listen_addr, "starting sequencer");
+        info_span!("Sequencer::run_until_stopped")
+            .in_scope(|| info!(config.listen_addr, "starting sequencer"));
```
Comment (Fraser999) [inline crates/astria-sequencer/src/sequencer.rs:128]:
```suggestion
        span.in_scope(|| info!(config.listen_addr, "starting sequencer"));
```

File: crates/astria-sequencer/src/sequencer.rs [hunk 5]
```diff
@@ -119,27 +124,29 @@ impl Sequencer {
...
+                        .in_scope(|| error!(err = e.as_ref(), "ABCI server exited with error"));
                 }
             }
             let _ = server_exit_tx.send(());
         });
 
         select! {
             _ = signals.stop_rx.changed() => {
-                info!("shutting down sequencer");
+                info_span!("Sequencer::run_until_stopped").in_scope(|| info!("shutting down sequencer"));
```
Comment (Fraser999) [inline crates/astria-sequencer/src/sequencer.rs:145]:
```suggestion
                span.in_scope(|| info!("shutting down sequencer"));
```

File: crates/astria-sequencer/src/sequencer.rs [hunk 6]
```diff
@@ -119,27 +124,29 @@ impl Sequencer {
...
             }
 
             _ = server_exit_rx => {
-                error!("ABCI server task exited, this shouldn't happen");
+                error_span!("Sequencer::run_until_stopped").in_scope(|| error!("ABCI server task exited, this shouldn't happen"));
```
Comment (Fraser999) [inline crates/astria-sequencer/src/sequencer.rs:149]:
```suggestion
                span.in_scope(|| error!("ABCI server task exited, this shouldn't happen"));
```

Comment (ethanoroshiba) [reply crates/astria-sequencer/src/sequencer.rs]: Made requested changes in 66e1b84 :)

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**